app/models/favorite_tag.rb

```diff
@@ -11,16 +11,24 @@ class FavoriteTag < ActiveRecord::Base
                                  }
 
   validate :within_limit, on: :create
-
-  after_save :expire_cached_home_favorite_tags
-  after_destroy :expire_cached_home_favorite_tags
-
   def within_limit
-    if user.favorite_tags(:reload).count >= ArchiveConfig.MAX_FAVORITE_TAGS
-      errors.add(:base, ts('Sorry, you can only save %{maximum} favorite tags.', maximum: ArchiveConfig.MAX_FAVORITE_TAGS))
+    if user && user.favorite_tags(:reload).count >= ArchiveConfig.MAX_FAVORITE_TAGS
+      errors.add(:base,
+                 ts("Sorry, you can only save %{maximum} favorite tags.",
+                    maximum: ArchiveConfig.MAX_FAVORITE_TAGS))
+    end
+  end
+
+  validate :canonical, on: :create
```

Comment:
We have code in place that destroys a favorite tag when it's de-canonized. That code shouldn't conflict with this -- it's currently tested here, and all tests are passing: https://github.com/otwcode/otwarchive/blob/master/features/tags_and_wrangling/favorite_tags.feature#L31